### PR TITLE
updated node engine to use 10.16.3

### PR DESCRIPTION
**Pull request by julien51**
# Description

Tests should fail based on @nickCuso's work:
https://github.com/unlock-protocol/unlock/pull/5006

One I know which one, I'll fix them one by one!

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [X] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by cnasc** [comment]
Looks good in general -- should we add an `engines` entry to the subrepos that don't have them?

**Comment by NickCuso** [comment]
This will pass.
The failure appears because these changes are not actually upgrading node.  When changing a call such as writeFileSync that's when you see the issue.

**Comment by julien51** [comment]
> Looks good in general -- should we add an engines entry to the subrepos that don't have them?

Ho, let me check if any is missing them!

**Comment by julien51** [comment]
> The failure appears because these changes are not actually upgrading node.
Can you tell me more?

**Comment by NickCuso** [comment]
With my previous PR, `npm run dev` fails with a callback error.  Updating the code to use `writeFileSync` works locally but causes CI to fail.

**Comment by NickCuso** [comment]
I just tried this branch locally.

`npm run test` has the following errors:
```
 FAIL  src/__tests__/utils/blog.test.js
  ● blog › should generate a blog feed array from a set of given markdown files

    expect(received).toEqual(expected) // deep equality

    Expected: "test2"
    Received: "test1"

      59 |     let feed = blog.generateBlogFeed('/foo/bar')
      60 |
    > 61 |     expect(feed[0].slug).toEqual('test2')
         |                          ^
      62 |     expect(feed[1].slug).toEqual('test1')
      63 |     expect(feed[0].__content).toEqual(undefined)
      64 |     expect(feed[1].__content).toEqual(undefined)

      at Object.toEqual (src/__tests__/utils/blog.test.js:61:26)

  ● blog › should generate a blog feed array from a set of given markdown files

    expect.assertions(4)

    Expected four assertions to be called but received one assertion call.

      55 |
      56 |   it('should generate a blog feed array from a set of given markdown files', () => {
    > 57 |     expect.assertions(4)
         |            ^
      58 |     const blog = require('../../utils/blog')
      59 |     let feed = blog.generateBlogFeed('/foo/bar')
      60 |

      at Object.assertions (src/__tests__/utils/blog.test.js:57:12)

  ● blog › should generate a blog index file from a blog index array

    expect(received).toEqual(expected) // deep equality

    Expected: "{\"items\":[{\"title\":\"This is a second sample post\",\"subTitle\":\"And some sample metadata\",\"publishDate\":\"Jan 7, 1979\",\"slug\":\"test2\"},{\"title\":\"This is a sample post\",\"subTitle\":\"And some sample metadata\",\"publishDate\":\"Dec 31, 1978\",\"slug\":\"test1\"}]}"
    Received: "{\"items\":[{\"slug\":\"test1\"},{\"slug\":\"test2\"}]}"

      103 |     blog.generateBlogIndexFile('/foo/bar', feed)
      104 |
    > 105 |     expect(writtenData).toEqual(blogJson)
          |                         ^
      106 |   })
      107 |
      108 |   it('should generate a blog RSS feed from a blog index array', () => {

      at Object.toEqual (src/__tests__/utils/blog.test.js:105:25)

  ● blog › should generate a blog RSS feed from a blog index array

    expect(received).toEqual(expected) // deep equality

    Expected: 2
    Received: 0

      120 |
      121 |     expect(xmlDoc.getElementsByTagName('channel').length).toEqual(1)
    > 122 |     expect(items.length).toEqual(2)
          |                          ^
      123 |     expect(items[0].querySelector('title').textContent).toEqual(
      124 |       'This is a second sample post'
      125 |     )

      at Object.toEqual (src/__tests__/utils/blog.test.js:122:26)

  ● blog › should generate a blog RSS feed from a blog index array

    expect.assertions(6)

    Expected six assertions to be called but received two assertion calls.

      107 |
      108 |   it('should generate a blog RSS feed from a blog index array', () => {
    > 109 |     expect.assertions(6)
          |            ^
      110 |
      111 |     const blog = require('../../utils/blog')
      112 |

      at Object.assertions (src/__tests__/utils/blog.test.js:109:12)
```

and `npm run dev` throws this:
```
fs.js:128
  throw new ERR_INVALID_CALLBACK();
  ^

TypeError [ERR_INVALID_CALLBACK]: Callback must be a function
    at maybeCallback (fs.js:128:9)
    at Object.writeFile (fs.js:1163:14)
    at generateBlogIndexFile (D:\unlock\unlock-protocol.com\src\utils\blog.js:65:6)
    at Object.<anonymous> (D:\unlock\unlock-protocol.com\scripts\build-blog.js:8:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
```

which is why my PR changed the writeFileSync call

**Comment by NickCuso** [comment]
I tested this branch on my linux machine as well.
npm run test passes
but npm run dev throws the same error as on windows.

**Comment by julien51** [comment]
Ok, let me look into that closely later today.

File: unlock-protocol.com/src/utils/blog.js [hunk 1]
```diff
@@ -60,12 +60,12 @@ const generatePostPages = postFeed => {
  * @param baseDir
  * @param postFeed
  */
-const generateBlogIndexFile = (baseDir, postFeed) => {
+const generateBlogIndexFile = (baseDir, postFeed, callback) => {
   // Write blog post index to output baseDirectory
   fs.writeFile(
     join(baseDir, 'blog', 'blog.index'),
     JSON.stringify({ items: postFeed }),
-    'utf8'
```
Comment (julien51) [inline unlock-protocol.com/src/utils/blog.js]: this was wrong.

**Comment by NickCuso** [comment]
`npm run dev` works now
`npm run test` still fails on windows, but that's likely a different issue entirely.

looks like we can merge this.  I still wonder if CI is actually using node 10 at this point though - I'm not clear why using writeFileSync wasn't also a valid solution.

**Comment by julien51** [comment]
I would gladly merge if @cla-bot would let me :/

**Pull request merged**